Sources/Public/Animation/LottieAnimationHelpers.swift

```diff
@@ -154,13 +154,21 @@ extension LottieAnimation {
   {
     switch strategy {
     case .legacyCodable:
-      return try JSONDecoder().decode(LottieAnimation.self, from: data)
+      do {
+        return try JSONDecoder().decode(LottieAnimation.self, from: data)
+      } catch {
+        return try DotLottieFile(data: data).mainAnimation()
+      }
     case .dictionaryBased:
-      let json = try JSONSerialization.jsonObject(with: data)
-      guard let dict = json as? [String: Any] else {
-        throw InitializableError.invalidInput
+      do {
+        let json = try JSONSerialization.jsonObject(with: data)
+        guard let dict = json as? [String: Any] else {
+          throw InitializableError.invalidInput
+        }
+        return try LottieAnimation(dictionary: dict)
+      } catch {
+        return try DotLottieFile(data: data).mainAnimation()
```

Comment:
Loading a `DotLottieFile` is a significantly more expensive operation that parsing a `LottieAnimation`, so I think this is potentially a surprising change. 

For example, I think it's probably acceptable in many circumstances to parse a `LottieAnimation` on the main thread, but I would never expect to decompress a zip file and perform other file operations on the main thread.

Since `.lottie` support is completely new functionality, I think it would be better to provide new public APIs specifically for loading `.lottie` files. These APIs should probably be asynchronous so the `.lottie` file decompression doesn't happen on the main thread.